lib/web/ui/server.go

```diff
@@ -101,38 +101,28 @@ func MakeServers(clusterName string, servers []types.Server) []Server {
 	return uiServers
 }
 
-// Kube describes a kube cluster.
-type Kube struct {
+// KubeCluster describes a kube cluster.
+type KubeCluster struct {
 	// Name is the name of the kube cluster.
 	Name string `json:"name"`
 	// Labels is a map of static and dynamic labels associated with an kube cluster.
 	Labels []Label `json:"labels"`
 }
 
 // MakeKubes creates ui kube objects and returns a list..
-func MakeKubes(clusterName string, servers []types.Server) []Kube {
-	kubeClusters := map[string]*types.KubernetesCluster{}
-
-	// Get unique kube clusters
-	for _, server := range servers {
-		// Process each kube cluster.
-		for _, cluster := range server.GetKubernetesClusters() {
-			kubeClusters[cluster.Name] = cluster
-		}
-	}
-
-	uiKubeClusters := make([]Kube, 0, len(kubeClusters))
-	for _, cluster := range kubeClusters {
+func MakeKubeClusters(clusters []types.KubeCluster) []KubeCluster {
+	uiKubeClusters := make([]KubeCluster, 0, len(clusters))
+	for _, cluster := range clusters {
 		uiLabels := []Label{}
```

Comment:
You could allocate the capacity for `uiLabels`
```suggestion
		staticLabels := cluster.GetStaticLabels()
		dynamicLabels := cluster.GetDynamicLabels()
		uiLabels := make([]Label, 0, len(staticLabels) + len(dynamicLabels)
```